test/shopify-cli/js_system_test.rb

```diff
@@ -122,5 +129,13 @@ def mock_kit_system(*input)
         .returns(mock(success?: true))
         .once
     end
+
+    def mock_system_call_with_capture(*input)
```

Comment:
This method actually creates an expectation and not just a mock. We should probably call it `expect_capture3_invoked` or something more explicit. Honestly though, I'd probably lose this method all together -- this expectation is the whole point of this test, and I feel it makes more sense to include the body of the expectation right in the test.